<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="image" default="/usb_cam/image_raw" />

  <!-- ROS parameter files -->
  <arg name="ros_param_file"             default="$(find scale_truck_control)/config/control.yaml"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="scale_truck_control" file="$(arg ros_param_file)"/>

  <!-- Start Scale Truck Control -->
  <node pkg="usb_cam" type="usb_cam_node" name="usb_cam" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap" />
  </node>

  <node pkg="scale_truck_control" type="scale_truck_control" name="scale_truck_control" output="screen" />

  <node pkg="rosserial_python" type="serial_node.py" name="serial_node" >
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baud" value="57600"/>
  </node>

</launch>
